support/vm/testing.go

```diff
@@ -0,0 +1,276 @@
+	// other expectations are optional
+	if ei.From != nil {
+		assert.Equal(t, *ei.From, invocation.Msg.from, "%s unexpected from address", identifier)
+	}
+	if ei.Value != nil {
+		assert.Equal(t, *ei.Value, invocation.Msg.value, "%s unexpected value", identifier)
+	}
+	if ei.Params != nil {
+		assert.True(t, ei.Params.matches(invocation.Msg.params), "%s params aren't equal (%v != %v)", identifier, ei.Params.val, invocation.Msg.params)
+	}
+	if ei.SubInvocations != nil {
+		for i, invk := range invocation.SubInvocations {
+			subidentifier := fmt.Sprintf("%s%d:", identifier, i)
+			require.Greater(t, len(ei.SubInvocations), i, "%s unexpected subinvocation [%s:%d]", subidentifier, invk.Msg.to, invk.Msg.method)
+			ei.SubInvocations[i].matches(t, subidentifier, invk)
+		}
+		missingInvocations := len(ei.SubInvocations) - len(invocation.SubInvocations)
+		if missingInvocations > 0 {
+			missingIndex := len(invocation.SubInvocations)
+			missingExpect := ei.SubInvocations[missingIndex]
+			require.Fail(t, "%s%d: expected invocation [%s:%d]", identifier, missingIndex, missingExpect.To, missingExpect.From)
+		}
+	}
+
+	// expect results
+	assert.Equal(t, ei.Exitcode, invocation.Exitcode, "%s unexpected exitcode", identifier)
+	if ei.Ret != nil {
+		assert.True(t, ei.Ret.matches(invocation.Ret), "%s unexpected return value (%v != %v)", identifier, ei.Ret, invocation.Ret)
+	}
+}
+
+func ParamsForInvocation(t *testing.T, vm *VM, idxs ...int) interface{} {
+	invocations := vm.Invocations()
+	var invocation *Invocation
+	for _, idx := range idxs {
+		require.Greater(t, len(invocations), idx)
+		invocation = invocations[idx]
+		invocations = invocation.SubInvocations
+	}
+	require.NotNil(t, invocation)
+	return invocation.Msg.params
+}
+
+//
+//  internal stuff
+//
+
+func initializeActor(ctx context.Context, t *testing.T, vm *VM, state runtime.CBORMarshaler, code cid.Cid, a address.Address, balance abi.TokenAmount) {
+	stateCID, err := vm.store.Put(ctx, state)
+	require.NoError(t, err)
+	actor := &TestActor{
+		Head:    stateCID,
+		Code:    code,
+		Balance: balance,
+	}
+	err = vm.setActor(ctx, a, actor)
+	require.NoError(t, err)
+}
+
+type addrPair struct {
+	pubAddr address.Address
+	idAddr  address.Address
+}
+
+func initAddressTree(t *testing.T, vm *VM, rootVerifier addrPair) cid.Cid {
+	m := adt.MakeEmptyMap(vm.store)
+	err := m.Put(adt.AddrKey(rootVerifier.pubAddr), cborAddrID(t, rootVerifier.idAddr))
+	require.NoError(t, err)
+
+	root, err := m.Root()
+	require.NoError(t, err)
+	return root
+}
+
+func cborAddrID(t *testing.T, idAddr address.Address) *cbg.CborInt {
+	id, err := address.IDFromAddress(idAddr)
+	require.NoError(t, err)
+	cbgId := cbg.CborInt(id)
+	return &cbgId
+}
+
+func verifregRootAddresses(t *testing.T) addrPair {
+	rootVerifierAddr, err := address.NewFromString("t3qfoulel6fy6gn3hjmbhpdpf6fs5aqjb5fkurhtwvgssizq4jey5nw4ptq5up6h7jk7frdvvobv52qzmgjinq")
+	require.NoError(t, err)
+
+	rootVerifierID, err := address.NewFromString("t080")
```

Comment:
Are we considering this a singleton actor? If so, we should declare it in the main actor code as a singleton. Please file an issue to follow up with that.

Reply:
Looking into it a bit, this actually should be a multisig actor, which means multiple account actors and an multisig actor need to be added to properly simulate adding verifiers. I don't think the multisig is a proper singleton in the sense that it needs a well known address, since it will only be addressed by its signers.

Since the VM doesn't check message signatures, it's currently possible to simply use this t080 address as the root verifier in the test VM. I'm dropping the root verifier account for now.